src/write.rs

```diff
@@ -897,7 +908,10 @@ fn write_local_file_header<T: Write>(writer: &mut T, file: &ZipFileData) -> ZipR
     // version needed to extract
     writer.write_u16::<LittleEndian>(file.version_needed())?;
     // general purpose bit flag
-    let flag = if !file.file_name.is_ascii() {
+    let flag = if !std::str::from_utf8(&file.file_name_raw)
+        .unwrap_or("あ")
```

Comment:
Hah, this took a moment to understand. A fun solution, but I'd prefer `file.file_name_raw.is_ascii()` ;)